astropy/units/core.py

```diff
@@ -1030,9 +1030,11 @@ def _is_equivalent(self, other, equivalencies=[]):
                     # to dimensionless (according to the equivalency)?
                     try:
                         (other / unit).decompose([a])
-                        return True
+
                     except Exception:
                         pass
+                    else:
+                        return True
```

Comment:
I think this change makes the logic *less* clear, so would prefer not to make it; it hides how `Exception` is meant to pass on to the next lines.

That said, the `pass` for the `Exception` is perhaps not the most logical here; if that is changed to `continue`, then I think it makes more sense: you try something and have two possible outcomes.